sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -826,13 +840,31 @@ public static String getCharsetDecoderErrorActionOnUnmappedCharacter() {
                     DEFAULT_CHARSET_DECODER_ERROR_ACTION_ON_UNMAPPED_CHARACTER));
     }
 
-    public static boolean isInsecureEmulatorConnectionAllowed() {
+    public static boolean isEmulatorHttpConnectionAllowed() {
         String httpForEmulatorAllowed = System.getProperty(
-            INSECURE_EMULATOR_CONNECTION_ALLOWED,
+            EMULATOR_HTTP_CONNECTION_ALLOWED,
             firstNonNull(
-                emptyToNull(System.getenv().get(INSECURE_EMULATOR_CONNECTION_ALLOWED_VARIABLE)),
-                String.valueOf(DEFAULT_INSECURE_EMULATOR_CONNECTION_ALLOWED)));
+                emptyToNull(System.getenv().get(EMULATOR_HTTP_CONNECTION_ALLOWED_VARIABLE)),
+                String.valueOf(DEFAULT_EMULATOR_HTTP_CONNECTION_ALLOWED)));
 
         return Boolean.parseBoolean(httpForEmulatorAllowed);
     }
+
+    public static boolean isEmulatorServerCertValidationDisabled() {
+        String certVerificationDisabledConfig = System.getProperty(
+            EMULATOR_CERTIFICATE_VALIDATION_DISABLED,
+            firstNonNull(
+                emptyToNull(System.getenv().get(EMULATOR_CERTIFICATE_VALIDATION_DISABLED_VARIABLE)),
+                String.valueOf(DEFAULT_EMULATOR_CERTIFICATE_VALIDATION_DISABLED)));
+
+        return Boolean.parseBoolean(certVerificationDisabledConfig);
+    }
+
+    public static String getEmulatorHost() {
+        return System.getProperty(
+            EMULATOR_HOST,
+            firstNonNull(
+                emptyToNull(System.getenv().get(EMULATOR_HOST_VARIABLE)),
+                DEFAULT_EMULATOR_HOST));
```

Comment:
either have the ,ocalhost logic being trigegred on localhost as emulatorHost here - or keep using null/empty and return "" here?

Reply:
changed to return empty string here